### PR TITLE
feat(Restore): single backup

### PR DESCRIPTION
**Pull request by mja9**
Created an overload for `Restore` that takes in a full `SingleBackup`. This can be used to restore databases that can be effectively recovered under the simple recovery model. Added a couple integration tests and verified they passed.

### CODE REVIEW

**Review by alexirion10 — APPROVED** [review]
🦃

**Review by slin5299 — APPROVED** [review]
lgtm!

File: src/SmoFacade/Server.cs [hunk 1]
```diff
@@ -168,10 +168,34 @@ public Database Database(string dbName)
...
+    public void Restore(SingleBackup fullBackup, string databaseName,
+      Func<int, TimeSpan> retryDurationProvider,
+      Func<string, string> fileRelocation = null)
+    {
+      var retryPolicy = Policy
+        .Handle<ExecutionFailureException>(e => e.InnerException != null
+                                                && e.InnerException is SqlException
+                                                && e.InnerException.Message
+                                                  .Contains("The process cannot access the file because it is being used by another process"))
+        .WaitAndRetry(10, retryDurationProvider);
```
Comment (AdityaNPL) [inline src/SmoFacade/Server.cs:180]: `retryPolicy` can be placed into a private variable and then re-used by both restore functions.
Maybe rename to `_restoreRetryPolicy`?

Comment (AdityaNPL) [reply src/SmoFacade/Server.cs:180]: looks like it needs `retryDurationProvider` (maybe a private utility method then)?

**Review by AdityaNPL — APPROVED** [review]
Makes sense, code LGTM!
Tests are sound too.

Just one small suggestion below

**Pull request merged**